src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,945 @@
+                        charSet = namedArg.Value.DecodeValue<CharSet>(SpecialType.System_Enum);
+                        break;
+
+                    case "SetLastError":
+                        // invalid values will be ignored
+                        setLastError = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "ExactSpelling":
+                        // invalid values will be ignored
+                        exactSpelling = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "PreserveSig":
+                        preserveSig = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "CallingConvention":
+                        // invalid values will be ignored
+                        callingConvention = namedArg.Value.DecodeValue<CallingConvention>(SpecialType.System_Enum);
+                        break;
+
+                    case "BestFitMapping":
+                        bestFitMapping = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "ThrowOnUnmappableChar":
+                        throwOnUnmappable = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+                }
+
+                position++;
+            }
+
+            if (!hasErrors)
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().SetDllImport(
+                    arguments.Index,
+                    moduleName,
+                    importName,
+                    DllImportData.MakeFlags(
+                        exactSpelling,
+                        charSet,
+                        setLastError,
+                        callingConvention,
+                        bestFitMapping,
+                        throwOnUnmappable),
+                    preserveSig);
+            }
+        }
+
+        internal sealed override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, DiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            Debug.Assert(!boundAttributes.IsDefault);
+            Debug.Assert(!allAttributeSyntaxNodes.IsDefault);
+            Debug.Assert(boundAttributes.Length == allAttributeSyntaxNodes.Length);
+            Debug.Assert(symbolPart == AttributeLocation.None || symbolPart == AttributeLocation.Return);
+
+            if (symbolPart != AttributeLocation.Return)
+            {
+                Debug.Assert(_lazyCustomAttributesBag != null);
+                Debug.Assert(_lazyCustomAttributesBag.IsDecodedWellKnownAttributeDataComputed);
+
+                if (ContainingType is { IsComImport: true, TypeKind: TypeKind.Class })
```

Comment:
>if (ContainingType is { IsComImport: true, TypeKind: TypeKind.Class }) [](start = 16, length = 70)

Should we enter this ```if``` for a local function? Please add a test.

Reply:
I think we shouldn't, because the diagnostic will be redundant (the containing method must be 'abstract' or 'extern', so there's nowhere to declare the local function)

---
In reply to: [354020269](https://github.com/dotnet/roslyn/pull/40011#discussion_r354020269) [](ancestors = 354020269)